### ISSUE TITLE
NullReferenceException in LiveChartsExtentions AsSegments

### ISSUE THREAD

**Issue by thomasjoscht** [beto-rodriguez/Live-Charts#98]
Hi, I'm determining a NullReferencException at:
   bei LiveCharts.LiveChartsExtentions.<AsSegments>d__1.MoveNext()
   bei LiveCharts.LineSeries.Plot(Boolean animate)
   bei LiveCharts.CoreComponents.Chart.UpdateSeries(Object sender, EventArgs e)
   bei System.Windows.Threading.DispatcherTimer.FireTick(Object unused)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

I don't know why this happens. I'm plotting a LineChart with frequently new dots (each second).

**Comment by beto-rodriguez** [comment]
Thank for reporting, I forgot to run unit test before publushing 0.6.5, I will try and let you know, are any of your points null? Are any of your series null? Are any of your series.values null?

**Comment by thomasjoscht** [comment]
I don't think so. I'm starting with an empty LineSeries and add iterativly new values to it.

**Comment by thomasjoscht** [comment]
Only for note: In another situation I get "n. def." ist kein gültiger Wert für die Eigenschaft "X1". ArgumentException at 
   bei System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)
   bei System.Windows.DependencyObject.SetValue(DependencyProperty dp, Object value)
   bei LiveCharts.CoreComponents.Chart.DrawAxes()
   bei LiveCharts.LineChart.DrawAxes()
   bei LiveCharts.LineChart.Scale()
   bei LiveCharts.CoreComponents.Chart.UpdateModifiedDataSeries(Object sender, EventArgs e)
   bei System.Windows.Threading.DispatcherTimer.FireTick(Object unused)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

I think this is not correlated with NullReferenceException, but I will notice it as information too.

**Comment by beto-rodriguez** [comment]
Are you using primitive types? I think that one of your setup function (X or Y) is null, they cannot be null, (I will force an excpetion when you set them to null, so you have more information about it and this do not happens again), I really need more information to solve this, but as far as I can see both expections are thrown where LiveCharts tries to use X and Y setup functions, I think one of them is null, or the eception is thrown in the function you setup. If I am not clear enough here is an example, please see what happens in your series collection configuration X and Y functions.

`var myConfig = new SeriesConfiguration<myType> ().X(//check this).Y(//also this)`

**Comment by thomasjoscht** [comment]
I've further investigated the behavior. I'm running a high multithreaded environment which actualises the series by a timer during ColletionChanged event of an ObservableCollection. The Plot() in LineSeries themes to run simultanously during changing values. In this scenario it happens that LineSeries Values.Points will become null (shortly it was an empty array). The extension method AsSegments is not aware and does not check NullReference and therefore the NullReferenceException will be thrown in foreach. I think it is necessary to check source in AsSegments to be not null like this:

```
        internal static IEnumerable<List<ChartPoint>> AsSegments(this IEnumerable<ChartPoint> source)
        {
            var buffer = new List<ChartPoint>();
            
            if (source != null)
            {
                foreach (var item in source)
                {
                    if (double.IsNaN(item.Y) || double.IsNaN(item.X))
                    {
                        yield return buffer;
                        buffer = new List<ChartPoint>();
                    }
                    else
                    {
                        buffer.Add(item);
                    }
                }
            }
            yield return buffer;
        }
```

I've tested it and it works.

The next problem with "n. def" comes at line 738 in DrawAxes() method of Chart.cs. I'm currently investigating why this happens.

**Comment by beto-rodriguez** [comment]
Nice and thanks, I will test and update with your research, thanks again

**Comment by beto-rodriguez** [comment]
About the second problem, it is a bit more complex than that, because if the chart is inverted that formula will not work, there is actually a unit test for this scenario, it was fixed for 0.6.4 I think I messed up something when publushing 0.6.5 I will see what is wrong there, but really thanks agian it halps a lot

**Comment by beto-rodriguez** [comment]
you were right about both problems, but the seconds scenario is practically impossible, I am really curious about how you got it...
